internal/controller/nonadminbackup_controller.go

```diff
@@ -97,11 +144,279 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+// that is referenced by the NameUUID within the NonAdminBackup (NAB) object.
+// This ensures the VeleroBackup is deleted before the NAB object itself is removed.
+//
+// Parameters:
+//   - ctx: Context to manage request lifetime.
+//   - logger: Logger instance for logging messages.
+//   - nab: Pointer to the NonAdminBackup object to be managed.
+//
+// This function first checks if the NAB object has the finalizer. If yes, it attempts to locate the associated
+// VeleroBackup object by the UUID stored in the NAB status. If a unique VeleroBackup object is found,
+// deletion is initiated, and the reconcile loop will be requeued. If multiple VeleroBackup objects
+// are found with the same label, the function logs an error and returns.
+//
+// Returns:
+//   - A boolean indicating whether to requeue the reconcile loop,
+//   - An error if VeleroBackup deletion or retrieval fails, for example, due to multiple VeleroBackup objects found with the same UUID label.
+func (r *NonAdminBackupReconciler) createVeleroDeleteBackupRequest(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	if !controllerutil.ContainsFinalizer(nab, constant.NabFinalizerName) ||
+		nab.Status.VeleroBackup == nil ||
+		nab.Status.VeleroBackup.NameUUID == constant.EmptyString {
+		return false, nil
+	}
+
+	// Initiate deletion of the VeleroBackup object only when the finalizer exists.
+	// For the ForceDelete we do not create DeleteBackupRequest
+	veleroBackupNameUUID := nab.Status.VeleroBackup.NameUUID
+	veleroBackup, err := function.GetVeleroBackupByLabel(ctx, r.Client, r.OADPNamespace, veleroBackupNameUUID)
+
+	if err != nil {
+		// Log error if multiple VeleroBackup objects are found
+		logger.Error(err, findSingleVBError, uuidString, veleroBackupNameUUID)
+		return false, err
+	}
+
+	if veleroBackup == nil {
+		logger.V(1).Info("VeleroBackup already deleted", "UUID", veleroBackupNameUUID)
+		return false, nil
+	}
+
+	deleteBackupRequest, err := function.GetVeleroDeleteBackupRequestByLabel(ctx, r.Client, r.OADPNamespace, veleroBackupNameUUID)
+	if err != nil {
+		// Log error if multiple DeleteBackupRequest objects are found
+		logger.Error(err, findSingleVDBRError, uuidString, veleroBackupNameUUID)
+		return false, err
+	}
+
+	if deleteBackupRequest == nil {
+		// Build the delete request for VeleroBackup created by NAC
+		deleteBackupRequest = builder.ForDeleteBackupRequest(r.OADPNamespace, "").
+			BackupName(veleroBackup.Name).
+			ObjectMeta(
+				builder.WithLabels(
+					velerov1.BackupNameLabel, label.GetValidName(veleroBackup.Name),
+					velerov1.BackupUIDLabel, string(veleroBackup.UID),
+					constant.NabOriginNameUUIDLabel, veleroBackupNameUUID,
+				),
+				builder.WithLabelsMap(function.GetNonAdminLabels()),
+				builder.WithAnnotationsMap(function.GetNonAdminBackupAnnotations(nab.ObjectMeta)),
+				builder.WithGenerateName(veleroBackup.Name+"-"),
+			).Result()
```

Comment:
It not easier to read if we create the YAML direct?

something like
```go
		deleteBackupRequest = &velerov1.DeleteBackupRequest{
			ObjectMeta: metav1.ObjectMeta{
				Namespace:    r.OADPNamespace,
				Labels:       function.GetNonAdminLabels(),
				Annotations:  function.GetNonAdminBackupAnnotations(nab.ObjectMeta),
				GenerateName: veleroBackup.Name + "-",
			},
			Spec: velerov1.DeleteBackupRequestSpec{
				BackupName: veleroBackup.Name,
			},
		}
```

Reply:
I really wanted to make this as similar to what Velero CLI does as possible so we don't introduce the `DeleteBackupRequest` that is unique to NAC only. We are taking what's in the Velero and adding on top our annotations and labels. Please check how Velero CLI is creating such object:
https://github.com/vmware-tanzu/velero/blob/6bffac5d0661ad066ea487774b3862b56067507d/pkg/cmd/cli/backup/delete.go#L124